Symfony/CS/Fixer/Symfony/NoEmptyStatementFixer.php

```diff
@@ -0,0 +1,222 @@
+     * - for, foreach, while (but not 'do - while')
+     * - switch
+     * - function (declaration, but not lambda)
+     * - declare (with '{' '}')
+     *
+     * @param Tokens $tokens
+     * @param int    $index           Semicolon index
+     * @param int    $curlyCloseIndex
+     */
+    private function fixSemicolonAfterCurlyBraceClose(Tokens $tokens, $index, $curlyCloseIndex)
+    {
+        static $beforeCurlyOpeningKinds = null;
+        if (null === $beforeCurlyOpeningKinds) {
+            $beforeCurlyOpeningKinds = array(T_ELSE, T_OPEN_TAG, ';');
+            if (defined('T_FINALLY')) {
+                $beforeCurlyOpeningKinds[] = T_FINALLY;
+            }
+        }
+
+        $curlyOpeningIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $curlyCloseIndex, false);
+        $beforeCurlyOpening = $tokens->getPrevMeaningfulToken($curlyOpeningIndex);
+
+        if ($tokens[$beforeCurlyOpening]->isGivenKind($beforeCurlyOpeningKinds) || $tokens[$beforeCurlyOpening]->equals(';')) {
+            $this->clear($tokens, $index);
+
+            return;
+        }
+
+        // check for class, interface and trait definitions
+        if ($tokens[$beforeCurlyOpening]->isGivenKind(T_STRING)) {
+            if ($this->isNonAnonymousClassClassyDefinition($tokens, $tokens->getPrevMeaningfulToken($beforeCurlyOpening))) {
+                $this->clear($tokens, $index);
+            }
+
+            return;
+        }
+
+        // early return check, below only control structures with conditions are fixed
+        if (!$tokens[$beforeCurlyOpening]->equals(')')) {
+            return;
+        }
+
+        $openingBrace = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $beforeCurlyOpening, false);
+        $beforeOpeningBrace = $tokens->getPrevMeaningfulToken($openingBrace);
+
+        if ($tokens[$beforeOpeningBrace]->isGivenKind(array(T_IF, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_SWITCH, T_CATCH, T_DECLARE))) {
+            $this->clear($tokens, $index);
+
+            return;
+        }
+
+        // check for function definition
+        if ($tokens[$beforeOpeningBrace]->isGivenKind(T_STRING)) {
+            $beforeString = $tokens->getPrevMeaningfulToken($beforeOpeningBrace);
+            if (null !== $beforeString && $tokens[$beforeString]->isGivenKind(T_FUNCTION)) {
+                $this->clear($tokens, $index); // implicit return
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function clear(Tokens $tokens, $index)
+    {
+        $tokens[$index]->clear();
+        // collapse whitespace if needed
+        if ($index === count($tokens) - 1 || !$tokens[$index - 1]->isWhitespace() || !$tokens[$index + 1]->isWhitespace()) {
+            return;
+        }
+
+        //  Note: isWhitespace also matches cleared tokens
+        for ($lower = 1; $tokens[$index - $lower]->isWhitespace(); ++$lower) {
+            //
+        }
```

Comment:
this will not be touched:

```php
for ($lower = 1; $tokens[$index - $lower]->isWhitespace(); ++$lower);
```

will be touched:

```php
for ($lower = 1; $tokens[$index - $lower]->isWhitespace(); ++$lower) {
   ; // this semicolon will be removed
```